apps/app/src/views/components/TreatmentOptionSearch.tsx

```diff
@@ -0,0 +1,124 @@
+import { Box, ChakraProvider, InputGroup, InputRightElement, Spinner } from '@chakra-ui/react';
+import { useEffect, useState } from 'react';
+import { gql, useApolloClient } from '@apollo/client';
+import { usePhoton } from '@photonhealth/react';
+import { useDebounce } from 'use-debounce';
+import { SelectField } from './SelectField';
+
+const SearchTreatmentOptionsQuery = gql`
+  query SearchTreatmentOptions($searchTerm: String!) {
+    treatmentOptions(searchTerm: $searchTerm) {
+      id: medicationId
+      form
+      name
+      ndc
+      route
+      strength
+      type
+      __typename
+    }
+  }
+`;
+
+type SelectedTreatment = {
+  medicineId: string;
+  name: string;
+};
+
+type TreatmentOptionSearchProps = {
+  setSelectedTreatment: (s: SelectedTreatment | undefined) => void;
+};
+
+export const TreatmentOptionSearch = ({
+  setSelectedTreatment: setSelectedTreatment
+}: TreatmentOptionSearchProps) => {
+  const client = usePhoton();
+  const [searchTerm, setSearchTerm] = useState<string>();
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<string>();
+
+  const [treatmentOptions, setTreatmentOptions] =
+    useState<Array<{ value: string; label: string }>>();
+
+  const clinicalClient = useApolloClient(client?.clinicalClient);
+
+  const [searchTermDebounce] = useDebounce(searchTerm, 800);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setLoading(true);
+      setError(undefined);
+
+      try {
+        const { data } = await clinicalClient.query({
+          query: SearchTreatmentOptionsQuery,
+          variables: { searchTerm: searchTermDebounce }
+        });
+
+        setTreatmentOptions(
+          data.treatmentOptions?.map((treatmentOption: { id: string; name: string }) => ({
+            value: treatmentOption.id,
+            label: treatmentOption.name
+          }))
+        );
+      } catch (err) {
+        setLoading(false);
+        setError(err as string);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    if (searchTermDebounce) {
+      fetchData();
+    }
+  }, [searchTermDebounce, clinicalClient]);
+
+  const handleSearchTermChange = (searchTerm: string) => {
+    setSearchTerm(searchTerm);
+  };
+
+  const handleSelectedTreatmentChange = (selectedTreatmentId: string) => {
```

Comment:
selecting something will only give us its value; we'll need to match it to our dataset if we want whoever uses this to be able to determine its name

Reply:
I'm not sure what you mean, you query comes back with id and name